testdata/lineage/basic-multiversion.txtar

```diff
@@ -0,0 +1,337 @@
+            init: "some string"
+        }
+    }
+},
+{
+    version: [0, 1]
+    schema: {
+        init:      string
+        optional?: int32
+    }
+    examples: {
+        withoutOptional: {
+            all:  42
+            init: "some string"
+        }
+        withOptional: {
+            all:  42
+            init: "some string"
+            optional: 32
+        }
+    }
+},
+{
+    version: [0, 2]
+    schema: {
+        init:        string
+        optional?:   int32
+        withDefault: *"foo" | "bar"
+    }
+    examples: {
+        withoutOptional: {
+            all:  42
+            init: "some string"
+            withDefault: "foo"
+        }
+        withOptional: {
+            all:  42
+            init: "some string"
+            optional: 32
+            withDefault: "bar"
+        }
+    }
+},
+{
+    version: [0, 3]
+    schema: {
+        init:        string
+        optional?:   int32
+        withDefault: *"foo" | "bar" | "baz"
+    }
+    examples: {
+        withoutOptional: {
+            all:  42
+            init: "some string"
+            withDefault: "baz"
+        }
+        withOptional: {
+            all:  42
+            init: "some string"
+            optional: 32
+            withDefault: "baz"
+        }
+    }
+},
+{
+    version: [1, 0]
+    schema: {
+        renamed:     string
+        optional?:   int32
+        withDefault: "foo" | *"bar" | "baz"
+    }
+    examples: {
+        withoutOptional: {
+            all:  42
+            renamed: "some string"
+            withDefault: "foo"
+        }
+        withOptional: {
+            all:  42
+            renamed: "some string"
+            optional: 32
+            withDefault: "bar"
+        }
+    }
+},
+{
+    version: [1, 1]
+    schema: {
+        renamed:     string
+        optional?:   int32
+        withDefault: "foo" | *"bar" | "baz" | "bing"
+    }
+    examples: {
+        withoutOptional: {
+            all:  42
+            renamed: "some string"
+            withDefault: "bing"
+        }
+        withOptional: {
+            all:  42
+            renamed: "some string"
+            optional: 32
+            withDefault: "bing"
+        }
+    }
+}]
+
+lenses: [{
+    to: [0, 3]
+    from: [1, 0]
+    input: _
+    result: {
+        init: input.renamed
+        all:  input.all
+        if (input.optional != _|_) {
+            optional: input.optional
+        }
+
+        withDefault: input.withDefault
+    }
+},
+{
+    to: [0, 1]
+    from: [0, 2]
+    input: _
+    result: {
+        init: input.init
+        all:  input.all
+        if (input.optional != _|_) {
+            optional: input.optional
+        }
+    }
+},
+{
+    to: [0, 2]
+    from: [0, 3]
+    input: _
+    result: {
+        init: input.init
+        all:  input.all
+        if (input.optional != _|_) {
+            optional: input.optional
+        }
+        // TODO: withDefault: input.withDefault doesn't work
+        withDefault: "foo"
```

Comment:
The use of `input.withDefault` does not only cause incompleteness on some of the test cases but also directly fails for some others. I guess we might want to explore that in a separate PR, but will probably require a more high-level discussion/decision.

Reply:
agreed, this is its whole own area for discussion